app/vmagent/main.go

```diff
@@ -322,11 +325,26 @@ func requestHandler(w http.ResponseWriter, r *http.Request) bool {
 			w.Write([]byte("OK"))
 		}
 		return true
+	default:
+		if strings.HasPrefix(r.URL.Path, "/static") {
+			file, err := staticFiles.ReadFile(strings.TrimPrefix(r.URL.Path, "/"))
```

Comment:
haven't checked that thoroughly, but looks like using `http.FileSystem` may simplify the logic. It also can autodetect headers based on content type, so you don't need to do it manually